runtime/parachains/src/paras/mod.rs

```diff
@@ -1345,6 +1361,9 @@ impl<T: Config> Pallet<T> {
 	) -> Weight {
 		let mut weight = 0;
 		for cause in causes {
+			weight += T::DbWeight::get().reads_writes(3, 2);
+			Self::deposit_event(Event::PvfCheckConcluded(*code_hash, cause.para_id(), true));
```

Comment:
Looks like this should be fixed in Substrate? So that `deposit_event` returns the weight instead of you are required to calculate this here manually?

Reply:
Yeah, I know. I did not want to open up this can of worms though. There are plenty examples of this, even in the polkadot repo, e.g.: 

https://github.com/paritytech/polkadot/blob/7311a3aa9f83730a079b0899c34bb813dd96c4e1/runtime/parachains/src/inclusion/mod.rs#L780

And it is not limited to `deposit_event` though. One of the worst cases IMO is the `Currency` trait where you can't even assume what the concrete implementation would do. 

IMO this problem should be addressed as a whole, holistically.

Anyway, I do not think that problem is within the scope of this particular PR (unless, you propose to remove the weight accounting for this case to make it inline with other cases found elsewhere). I see a solution should be first discussed in an issue in Substrate (cc @shawntabrizi), not in this thread. As soon as we have a solution though, I'd be down to migrate this code (and the rest of polkadot's for that matter) to the new mechanism.

Reply:
Yeah, I also don't wanted you to fix this as part of this pr.